external/js/cothority/src/byzcoin/byzcoin-rpc.ts

```diff
@@ -172,9 +177,21 @@ export default class ByzCoinRPC implements ICounterUpdater {
             transaction,
             version: currentVersion,
         });
+        const counters = this.counters();
 
         // The error might be in the response, so we look for it and then reject the promise.
         const resp = await this.conn.send(req, AddTxResponse) as AddTxResponse;
+        // If the transaction has been successful, we update all cached counters. If the transaction
+        // failed, all involved counters are reset and will have to be fetched again.
+        transaction.instructions.forEach((instruction) => {
+            instruction.signerIdentities.forEach((signer, i) => {
+                if (resp.error.length === 0) {
+                    counters.set(signer.toString(), instruction.signerCounter[i]);
+                } else {
+                    counters.delete(signer.toString());
```

Comment:
Does this mean that if a signer counter error occurred, it will remove the counter thus the next try will fetch from a conode ?

Reply:
yes - very simple cache management: if something goes wrong, clear the state and start from scratch.